interfaces/dbus/spec.go

```diff
@@ -21,16 +21,50 @@ package dbus
 
 import (
 	"bytes"
+	"fmt"
 	"sort"
 
 	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/snap"
 )
 
 // Specification keeps all the dbus snippets.
 type Specification struct {
 	// Snippets are indexed by security tag.
 	snippets     map[string][]string
 	securityTags []string
+
+	// sessionServices map a security tag to the dbus-serivce file content
+	sessionServices map[string]string
+}
+
+// AddService adds a new dbus service
+func (spec *Specification) AddService(bus, name string, appInfo *snap.AppInfo) {
+	if bus == "session" {
+		if spec.sessionServices == nil {
+			spec.sessionServices = make(map[string]string)
+		}
+		// We set only 'Name' and 'Exec' for now. We may add
+		// 'User' for 'system' services when we support
+		// per-snap users. Don't specify 'SystemdService' and
+		// just let dbus-daemon launch the service since
+		// 'SystemdService' is only used by dbus-daemon to
+		// tell systemd to launch the service and systemd user
+		// sessions aren't available everywhere yet.
+		spec.sessionServices[appInfo.SecurityTag()] = fmt.Sprintf(`[D-BUS Service]
+Name=%s
+Exec=%s
+`, name, appInfo.LauncherCommand())
```

Comment:
Note that we have `systemd.Service` and the `systemd` backend as well. We could perhaps reuse some of those here.